opentelemetry-api/src/opentelemetry/util/_decorator.py

```diff
@@ -15,18 +15,20 @@
 import asyncio
 import contextlib
 import functools
-import typing
-from typing import Callable, Generic, Iterator, TypeVar
+from typing import TYPE_CHECKING, Callable, Generic, Iterator, TypeVar
 
 V = TypeVar("V")
 R = TypeVar("R")  # Return type
 Pargs = TypeVar("Pargs")  # Generic type for arguments
 Pkwargs = TypeVar("Pkwargs")  # Generic type for arguments
 
-if hasattr(typing, "ParamSpec"):
-    # only available in python 3.10+
-    # https://peps.python.org/pep-0612/
-    P = typing.ParamSpec("P")  # Generic type for all arguments
+# We don't actually depend on typing_extensions but we can use it in CI with this conditional
+# import. ParamSpec can be imported directly from typing after python 3.9 is dropped
+# https://peps.python.org/pep-0612/.
```

Comment:
Should we just add the `typing-extensions` dependency? I suppose it's possible that users of this library will get failures here and need to install `typing-extensions` themselves.

Reply:
I think it would be unnecessary. Bot mypy and pyright PyPI packages already pull in typing-extensions as dependency.